### DISCUSSION TITLE
SpeedDial.Action is not typed correctly?

### DISCUSSION THREAD

**Discussion by lucksp** [react-native-elements/react-native-elements#3750] [Q&A]
Using `@rneui/themed@4.0.0-rc.1` via `github:react-native-elements/react-native-elements#themed` install.  
I am adding a `SpeedDial` and with the `SpeedDial.Action` component , as a `child` of the parent.  

```
<SpeedDial
          isOpen={speedDialOpen}
          icon={{ name: 'info', color: '#fff' }}
          openIcon={{ name: 'close', color: '#fff' }}
          onOpen={() => setSpeedDialOpen(prev => !prev)}
          onClose={() => setSpeedDialOpen(prev => !prev)}
        >
          <SpeedDial.Action
            icon={{ name: 'auto-delete', color: '#fff' }}
            title="Clear Stored Map"
            onPress={handleClearStoredMap}
          />
        </SpeedDial>
```

When adding the Action child, Typescript yells about:

> No overload matches this call.
  Overload 1 of 2, '(props: PropsWithChildren<SpeedDialProps>, context?: any): ReactElement<any, any> | null', gave the following error.
    Type 'Element' is not assignable to type '(ReactChild[] & (boolean | ReactChild | ReactFragment | ReactPortal | null)) | undefined'.
      Type 'ReactElement<any, any>' is not assignable to type 'ReactChild[] & ReactPortal'.
        Type 'ReactElement<any, any>' is missing the following properties from type 'ReactChild[]': length, pop, push, concat, and 29 more.
  Overload 2 of 2, '(props: SpeedDialProps): ReactElement<any, any> | null', gave the following error.
    Type 'ReactElement<any, any>' is missing the following properties from type 'ReactChild[]': length, pop, push, concat, and 29 more.ts(2769)
SpeedDial.d.ts(12, 5): The expected type comes from property 'children' which is declared here on type 'IntrinsicAttributes & SpeedDialProps & { children?: ReactNode; }'
SpeedDial.d.ts(12, 5): The expected type comes from property 'children' which is declared here on type 'IntrinsicAttributes & SpeedDialProps'
No quick fixes available

<img width="660" alt="image" src="https://user-images.githubusercontent.com/15162169/217322694-685ee71e-d4cc-4a84-b4fd-aedfbeacda0b.png">

Is there something wrong with my code or is it the lib?  I am using version:

**Comment by minhvu7896** [comment]
Yeah, When I use js, its okay, but with tsx I get same error.

**Comment by WebDevGenie** [comment]
It's because the SpeedDialProps expects multiple children. If you have two `<SpeedDial.Action />`s then it will get rid of the error.
![image](https://github.com/react-native-elements/react-native-elements/assets/54092140/93a7d3aa-881e-4ecd-ac6d-ac2152107271)

![image](https://github.com/react-native-elements/react-native-elements/assets/54092140/be7d6aab-283f-436f-88b0-63c71b747fed)
